Source/ProgressionSystemRuntime/Public/Widgets/PSMenuWidget.h

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "Blueprint/UserWidget.h"
+#include "PSMenuWidget.generated.h"
+enum class ECurrentGameState : uint8;
+
+
+/**
+ * Widget to display the progression as stars in the main menu.
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSMenuWidget : public UUserWidget
+{
+	GENERATED_BODY()
+
+public:
+	// Horizontal Box widget for storing stars
+	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UHorizontalBox> HorizontalBox = nullptr;
+
+	// Background overlay for tint effect
+	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UImage> PSCBackgroundOverlay = nullptr;
+
+	// Background overlay lock icon
+	UPROPERTY(EditDefaultsOnly, BlueprintReadWrite, Category = "C++", meta = (BlueprintProtected, BindWidget))
+	TObjectPtr<class UImage> PSCBackgroundIconLock = nullptr;
+
+	// Function to add images to the Horizontal Box
+	UFUNCTION(BlueprintCallable)
+	void AddImagesToHorizontalBox(int AmountOfUnlockedPoints, int AmountOfLockedPoints);
```

Comment:
Use only `int32` instead of `int`. It's crucial in Unreal Engine to guarantee consistent behavior and size across various platforms, enhancing code reliability and cross-platform compatibility.

Reply:
done